### PR TITLE
Dat 49582 goals chart

### PR DESCRIPTION
**Pull request by crazypixel**
I want to push this component and let @portons to play with it.
I have 2 remaining tasks - 
1. add margins (Y axis)
2. add initial value for the drag line. or controlled.

### CODE REVIEW

File: app/src/components/base/charts/GoalsChart.js [hunk 1]
```diff
@@ -255,78 +278,81 @@ const GoalsChart = ({
...
-              padding +
-              state.translation -
-              4}`,
-            'l 4, -4',
-            'l 4, 4',
-            'z'
-          ].join(' ')}
-        />
-      </DragSvg>
+      {!!state.height && (
```
Comment (itaydafna-datorama) [inline app/src/components/base/charts/GoalsChart.js]: height has to be extracted from `Container` `ref` in order for the proper `initialTranslation` value to be passed to `DragSvg`

File: app/src/components/base/charts/GoalsChart.js [hunk 2]
```diff
@@ -210,12 +230,15 @@ const GoalsChart = ({
     }
   }, [el]);
 
-  const percentage =
-    minY[1] +
-    Math.round(
-      ((-1 * state.translation) / (state.height - 2 * padding)) *
-        (maxY[1] - minY[1])
-    );
+  const getTranslation = percentage =>
```
Comment (Chikel) [inline app/src/components/base/charts/GoalsChart.js]: Where is this algorithm coming from? Couldn't find the origin.

Comment (itaydafna-datorama) [reply app/src/components/base/charts/GoalsChart.js]: I had to figure it out. 
It's based on how percentage was calculated based on the `state.translation` (look at modified code to the left).
So I played around with this "equation" and managed to get the opposite (translation based on percentage)

**Comment by itaydafna-datorama** [comment]
here are fun mocks for testing the label renderers in case anyone is interested :)

```
const axisLabelRenderer = ({key,x,y,value}) => (
    <text
        key={key}
        alignmentBaseline="middle"
        textAnchor="end"
        x={50 + x}
        y={y}
        fill="green"
        style={{fontWeight: 500, fontSize: 11}}
    >
        🍇{value}🍉
    </text>);


const valueLabelRenderer = ({x,y,value})=> (<text
    alignmentBaseline="middle"
    textAnchor="end"
    x={50 + x}
    y={y}
    fill="orange"
    style={{fontWeight: 500, fontSize: 13}}
>
    {value} 
</text>);

const lineLabelRenderer = ({x,y,value})=> (<text
    alignmentBaseline="middle"
    textAnchor="end"
    x={x}
    y={y}
    fill="purple"
    style={{fontWeight: 500, fontSize: 11}}
>
    {value} 😻
</text>);
```

**Pull request merged**